src/components/DatePicker/DatePicker.tsx

```diff
@@ -384,17 +329,19 @@ export const DatePicker = ({
         disabled={disabled}
         {...props}
       >
-        {ModeSelector && (
-          <ModeSelector
-            mode={mode}
-            allowedModes={allowedModes ?? [mode]}
-            onChangeMode={onChangeModeInternal}
-          />
-        )}
-        {mode === 'dayRangePicker' && (
+        {(ModeSelector && pickerMode !== 'timePicker')
+          ? (
+            <ModeSelector
+              mode={pickerMode}
+              allowedModes={allowedModes ?? [pickerMode]}
```

Comment:
Does it make sense to have a mode selector with a single allowed mode? Not obvious to me why we'd render the mode selector if there are less than 2 allowed modes.

I see this pre-dated this PR, so ignore if it feels like a side convo